### PR TITLE
OLS-505: Refactor script scripts/question_gen.py

### PR DESCRIPTION
**Pull request by Nehanth**
## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: scripts/questions_gen.py [hunk 1]
```diff
@@ -59,25 +59,27 @@ def eval_parser(eval_response: str):
...
-            reasoning = eval_response_parsed[1]
-            return score, reasoning
-        case 3:
-            score_str, reasoning_str = eval_response_parsed[1], eval_response_parsed[2]
-            score = float(score_str) if score_str else 0
-            reasoning = reasoning_str.lstrip("\n")
-            return score, reasoning
-        case _:
-            return 0, eval_response
+    no_data_lines = 0
```
Comment (tisnik) [inline scripts/questions_gen.py:62]: prob. use enum there

File: scripts/questions_gen.py [hunk 2]
```diff
@@ -59,25 +59,27 @@ def eval_parser(eval_response: str):
...
+    single_lines = 1
+    double_lines = 2
+    triple_lines = 3
+
+    if eval_len == no_data_lines:
```
Comment (tisnik) [inline scripts/questions_gen.py:67]: why was match-case removed? Is not it much more readable?

File: scripts/questions_gen.py [hunk 3]
```diff
@@ -206,58 +255,72 @@ async def questions_eval(
...
+        "response": summary.response,
+        "ref": referenced_documents,
+        "ref_doc": summary.source_nodes[0].text,
+        "ref_doc_score": summary.source_nodes[0].score,
+        "passing": result.passing,
+        "feedback": result.feedback,
+        "score": result.score,
+    }
+
+    # TODO: OLS-619 Evaluation model can return `None` instead of score, so
```
Comment (tisnik) [inline scripts/questions_gen.py:309]: the TODO was fixed there, so might be removed

**Review by tisnik — COMMENTED** [review]
some nitpicks, but the refactored code looks better than the original

File: scripts/questions_gen.py [hunk 4]
```diff
@@ -158,20 +195,16 @@ def generate_summary(
                         new += f"   - {k}: {v}\n"
                 value = new
```
Comment (tisnik) [inline scripts/questions_gen.py:196]: this is the problematic line you need to resolve -> the `new` might be unbound

File: scripts/questions_gen.py (hunk 2 above)
Comment (Nehanth) [reply scripts/questions_gen.py:67]: says this in the story - commented/named magic constants in eval_parser

Comment (tisnik) [reply scripts/questions_gen.py:67]: then just use named constants in pattern matcher ;)

File: scripts/questions_gen.py [hunk 5]
```diff
@@ -60,18 +70,16 @@ def eval_parser(eval_response: str):
...
+        case EvalResponseLength.SINGLE_LINES:
+            return 0, eval_response_parsed[0]
+        case EvalResponseLength.DOUBLE_LINES:
             score_str = eval_response_parsed[0]
             score = float(score_str) if score_str else 0
             reasoning = eval_response_parsed[1]
             return score, reasoning
-        case 3:
+        case EvalResponseLength.TRIPLE_LINES:
             score_str, reasoning_str = eval_response_parsed[1], eval_response_parsed[2]
```
Comment (tisnik) [inline scripts/questions_gen.py:83]: nice!

File: scripts/questions_gen.py [hunk 6]
```diff
@@ -134,44 +156,61 @@ def generate_summary(
...
-        file.write(json_metadata)
+        file.write(markdown_content)
+
 
+def generate_full_results_markdown(full_results):
+    """Generate markdown content for full results."""
     full_results_markdown_content = "    \n"
     for res in full_results:
         for key, value in res.items():
             if isinstance(value, list) and isinstance(value[0], dict):
```
Comment (tisnik) [inline scripts/questions_gen.py:197]: prob worth skip empty list?

**Review by tisnik — COMMENTED** [review]
one nitpick, look ok instead

**Pull request merged**